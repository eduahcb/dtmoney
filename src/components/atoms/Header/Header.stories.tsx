import React, { ReactElement } from 'react'
import { action } from '@storybook/addon-actions'

import { Brand } from 'components/atoms/Brand'
import { Header } from 'components/atoms/Header/Header'

export default {
  title: 'Components/Header',
  component: Header
}

type UsageProps = {
  title: string
}

export const usage = ({ title }: UsageProps): ReactElement => (
  <Header>
    <h1>{title}</h1>
  </Header>
)

usage.args = {
  title: 'Title'
}

export const withBrand = (): ReactElement => (
  <Header>
    <Brand />
  </Header>

)

export const withButton = (): ReactElement => (
  <Header>
    <Brand />
    <button onClick={action('clicked')}>click here</button>
  </Header>
)
