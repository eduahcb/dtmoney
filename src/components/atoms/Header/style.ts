import styled from 'styled-components'

export const Container = styled.header`
  background-color: var(--blue);
  height: 124px;
`

export const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  max-width: 1120px;
  margin: 0 auto;
  
  padding: 32px 16px;
`
