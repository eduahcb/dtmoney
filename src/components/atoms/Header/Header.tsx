import React, { ReactElement, ReactNode } from 'react'

import { Container, Content } from './style'

type HeaderProps = {
  children: ReactNode
}

export const Header = ({ children }: HeaderProps): ReactElement => {
  return (
    <Container>
      <Content>
        {children}
      </Content>
    </Container>
  )
}
