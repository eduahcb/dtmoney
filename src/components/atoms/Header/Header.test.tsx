import React from 'react'

import { render, screen } from '@testing-library/react'

import { Header } from 'components/atoms/Header'

describe('Header', () => {
  test('renders children', () => {
    render(<Header><p>this is a text</p></Header>)

    expect(screen.getByText('this is a text')).toBeInTheDocument()
  })

  test('match snapshot', () => {
    const { asFragment } = render(<Header><p>this is a text</p></Header>)

    expect(asFragment()).toMatchSnapshot()
  })
})
