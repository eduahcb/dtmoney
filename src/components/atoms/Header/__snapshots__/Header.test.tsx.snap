// Vitest Snapshot v1

exports[`Header > match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: var(--blue);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 1120px;
  margin: 0 auto;
  padding: 32px 16px 192px;
}

<header
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        this is a text
      </p>
    </div>
  </header>
</DocumentFragment>
`;
